xbmc/filesystem/CurlFile.cpp

```diff
@@ -1959,6 +1960,8 @@ const std::string CCurlFile::GetProperty(XFILE::FileProperty type, const std::st
     return m_state->m_httpheader.GetCharset();
   case FILE_PROPERTY_MIME_TYPE:
     return m_state->m_httpheader.GetMimeType();
+  case FILE_PROPERTY_RESPONSE_CODE:
+    return std::to_string(m_httpresponse);
```

Comment:
why does this need to be a string?

Reply:
Because of the return value (string) of the function